predicate/parser.go

```diff
@@ -201,3 +223,13 @@ scanRegularTagValue:
 		}
 	}
 }
+
+// peekRune returns the next rune that would be read by the scanner.
+func (p *parser) peekTok() influxql.Token {
+	tok, _, _ := p.scanIgnoreWhitespace()
+	if tok != influxql.EOF {
```

Comment:
curious why there is no unscan when its `EOF`?

Reply:
the scan will consume the current position until eof.